### PR TITLE
Cypress/E2E: Fix channel sidebar specs

### PR DESCRIPTION
**Pull request by josephbaylon**
#### Summary
- Added `ui/modal.js` and `ui/modal.d.ts`
- Added `uiCloseModal` and `uiCloseWhatsNewModal` commands
- Closed "What's new" modal on affected tests

#### Ticket Link
None - both on master and release-5.26

Screen Shot 2020-07-23 at 1 35 03 PM
Screen Shot 2020-07-23 at 1 36 20 PM
Screen Shot 2020-07-23 at 1 37 02 PM
Screen Shot 2020-07-23 at 1 37 46 PM
Screen Shot 2020-07-23 at 1 38 44 PM
Screen Shot 2020-07-23 at 1 40 09 PM

### CODE REVIEW

File: e2e/cypress/integration/channel_sidebar/category_collapsing_unread_filter_spec.js [hunk 1]
```diff
@@ -37,6 +37,9 @@ describe('Channel sidebar', () => {
     });
 
     it('should display collapsed state when collapsed', () => {
+        // # Close "What's new" modal
+        cy.uiCloseWhatsNewModal();
```
Comment (saturninoabril) [inline e2e/cypress/integration/channel_sidebar/category_collapsing_unread_filter_spec.js:41]: I'm good as is but better if this is placed in the before hook so each test can be run with `.only` when have to in the future.

File: e2e/cypress/integration/channel_sidebar/drag_and_drop_spec.js [hunk 2]
```diff
@@ -36,13 +36,16 @@ describe('Channel sidebar', () => {
...
             cy.visit(`/${team.name}/channels/town-square`);
-
-            // * Verify that we've switched to the new team
-            cy.get('#headerTeamName', {timeout: TIMEOUTS.ONE_MIN}).should('be.visible').should('contain', teamName);
         });
     });
 
     it('should move channel to correct place when dragging channel within category', () => {
+        // # Close "What's new" modal
+        cy.uiCloseWhatsNewModal();
```
Comment (saturninoabril) [inline e2e/cypress/integration/channel_sidebar/drag_and_drop_spec.js:44]: Same here.

File: e2e/cypress/support/ui/modal.d.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+        /**
+         * Close "What's new" modal
+         *
+         * @example
+         *   cy.uiCloseWhatsNewModal();
+         */
+        uiCloseWhatsNewModal(): Chainable;
+    }
+}
```
Comment (saturninoabril) [inline e2e/cypress/support/ui/modal.d.ts:38]: 🎉

**Comment by saturninoabril** [comment]
Automatic cherry-pick didn't work. Will do it manually.

File: e2e/cypress/integration/channel_sidebar/category_collapsing_unread_filter_spec.js (hunk 1 above)
Comment (josephbaylon) [reply e2e/cypress/integration/channel_sidebar/category_collapsing_unread_filter_spec.js:41]: Reason is page is loaded in beforeEach and modal only appear only once. That would require restructuring the whole test

File: e2e/cypress/integration/channel_sidebar/drag_and_drop_spec.js (hunk 2 above)
Comment (josephbaylon) [reply e2e/cypress/integration/channel_sidebar/drag_and_drop_spec.js:44]: Same reason as above

**Pull request merged**